src/wallet/wallet.cpp

```diff
@@ -1775,9 +1833,21 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
             WalletLogPrintf("Still rescanning. At block %d. Progress=%f\n", block_height, progress_current);
         }
 
+        bool skip_block = false;
+        if (fast_rescan_filter) {
+            fast_rescan_filter->UpdateIfNeeded();
+            auto match_result = fast_rescan_filter->MatchesBlock(block_hash);
+            if (match_result.has_value() && !match_result.value()) {
+                // Block is skipped, pretend that we found it successfully
+                result.last_scanned_block = block_hash;
+                result.last_scanned_height = block_height;
+                skip_block = true;
+            }
+        }
+
         // Read block data
         CBlock block;
-        chain().findBlock(block_hash, FoundBlock().data(block));
+        if (!skip_block) chain().findBlock(block_hash, FoundBlock().data(block));
```

Comment:
After some consideration, I decided to keep as it is for now to keep the code simpler, as I think default-ctoring a `CBlock` is neglectible here compared to other operations (didn't do any benchmarking or look deeper into it though). Happy to follow-up here though, will keep this conversation open. (Maybe also a topic to discuss for PR review club tomorrow, in case there is some time at the end?).